From 021c7cf8b270571f8626200f557ac9074c7218b2 Mon Sep 17 00:00:00 2001
From: Peter Ujfalusi <peter.ujfalusi@ti.com>
Date: Fri, 17 Jan 2014 13:51:22 +0200
Subject: [PATCH 161/188] ARM: dra7: hwmod: Correct mcasp3 domain (to l4_per2)

This commit comes from branch ti-linux-3.12.y:
  git://git.ti.com/ti-linux-kernel/ti-linux-kernel.git

McASP3 is in L4_PER2 domain and not in L3_MAIN_1.

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@ti.com>
(cherry picked from commit dfd88356a6be344d3e78f26272b9ca2b77523623)
---
 arch/arm/mach-omap2/omap_hwmod_7xx_data.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-omap2/omap_hwmod_7xx_data.c b/arch/arm/mach-omap2/omap_hwmod_7xx_data.c
index f01a150..5ae0ede 100644
--- a/arch/arm/mach-omap2/omap_hwmod_7xx_data.c
+++ b/arch/arm/mach-omap2/omap_hwmod_7xx_data.c
@@ -2950,9 +2950,9 @@ static struct omap_hwmod_addr_space dra7xx_mcasp3_addrs[] = {
 	{ }
 };
 
-/* l3_main_1 -> mcasp3 */
-static struct omap_hwmod_ocp_if dra7xx_l3_main_1__mcasp3 = {
-	.master		= &dra7xx_l3_main_1_hwmod,
+/* l4_per2 -> mcasp3 */
+static struct omap_hwmod_ocp_if dra7xx_l4_per2__mcasp3 = {
+	.master		= &dra7xx_l4_per2_hwmod,
 	.slave		= &dra7xx_mcasp3_hwmod,
 	.clk		= "l3_iclk_div",
 	.addr		= dra7xx_mcasp3_addrs,
@@ -3698,7 +3698,7 @@ static struct omap_hwmod_ocp_if *dra7xx_hwmod_ocp_ifs[] __initdata = {
 	&dra7xx_dsp2__l3_main_1,
 	&dra7xx_l3_main_1__hdmi,
 	&dra7xx_l3_main_1__aes,
-	&dra7xx_l3_main_1__mcasp3,
+	&dra7xx_l4_per2__mcasp3,
 	&dra7xx_l4_per1__elm,
 	&dra7xx_l4_wkup__gpio1,
 	&dra7xx_l4_per1__gpio2,
-- 
1.7.5.4

